src/html_parser/readability.py

```diff
@@ -32,8 +32,14 @@ def parse(self, input: ParserInput) -> ParserOutput:
         :param url: URL of web page
 
         :return ParsedHTML: parsed HTML
+        :raise ValueError: input contains a null value for URL
         """
 
+        if input.document_url is None:
+            raise ValueError(
```

Comment:
Deployed in the pipeline this will cause the process to stop, is it possible to log this out for now and skip the file? Eventually we can add this to a failed documents list that we then handle post run?

Reply:
The parser processes all documents without a url and content type separately - this is more just a statement in the individual pdf parsing method so that it satisfies type constraints (otherwise `requests.get` in the next line would fail anyway)